### PR TITLE
Added tool for generating boilerplate react component + test code

### PR DESCRIPTION
**Pull request by devYaoYH**
- Create python script tool to generate boilerplate react component + basic snapshot test code
- Registered script to run with `yarn create-component` in client
- Moved `EAN_chksum.py` script into `tools/` folder at directory root

### CODE REVIEW

File: tools/create_component.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    opt = input("Folder exists, confirm potential overwrite(y/n):")
+    if (opt != "y"):
+        print("Script aborted, folder already exists")
+        exit(1)
+else:
+    os.makedirs(component_folder);
+
+# Read template files
+__dir__ = os.getcwd();
+template_component = "template.tsx"  #TODO: Replace with read from crc.config.json
```
Comment (BhavyaLight) [inline tools/create_component.py:43]: use all caps for constants 
```suggestion
COMPONENTS_TEMPLATE_PATH = "template.tsx"  #TODO: Replace with read from crc.config.json
```

File: tools/create_component.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+template_test = "template.test.tsx"  #TODO: Replace with read from crc.config.json
```
Comment (BhavyaLight) [inline tools/create_component.py:44]:
```suggestion
TEST_TEMPLATE_PATH = "template.test.tsx"  #TODO: Replace with read from crc.config.json
```

File: tools/create_component.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        for k in var_mapping.keys():
+            line = line.replace(k,var_mapping[k])
+        component += line
+# Read test template
+test = ""
+with open(os.path.join(__dir__,template_test),'r') as fin:
+    for line in fin:
+        for k in var_mapping.keys():
+            line = line.replace(k,var_mapping[k])
+        test += line
```
Comment (BhavyaLight) [inline tools/create_component.py:62]: consider abstracting this to a function and re-using as it is repetitive

File: tools/create_component.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+# Write files
+component_file = os.path.join(component_folder,args.name+".tsx")
```
Comment (BhavyaLight) [inline tools/create_component.py:65]: nit: this is the file path
```suggestion
component_file_path = os.path.join(component_folder,args.name+".tsx")
```

File: tools/create_component.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+test_file = os.path.join(component_folder,args.name+".test.tsx")
+index_file = os.path.join(component_folder,"index.tsx");
+with open(os.path.join(__dir__,component_file),'w+') as fout:
+    for line in component:
+        fout.write(line)
+with open(os.path.join(__dir__,test_file),'w+') as fout:
+    for line in test:
+        fout.write(line)
```
Comment (BhavyaLight) [inline tools/create_component.py:73]: same for writing a function for this

File: tools/create_component.py [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+with open(os.path.join(__dir__,index_file),'w+') as fout:
+    fout.write("import {0} from \"./{0}\";\n".format(args.name));
+    fout.write("export default {};\n".format(args.name));
```
Comment (BhavyaLight) [inline tools/create_component.py:76]: why not make a template for index too?

File: tools/create_component.py (hunk 1 above)
Comment (devYaoYH) [reply tools/create_component.py:43]: updated

File: tools/create_component.py (hunk 2 above)
Comment (devYaoYH) [reply tools/create_component.py:44]: updated

File: tools/create_component.py (hunk 3 above)
Comment (devYaoYH) [reply tools/create_component.py:62]: abstracted into function

File: tools/create_component.py (hunk 4 above)
Comment (devYaoYH) [reply tools/create_component.py:65]: renamed

File: tools/create_component.py (hunk 5 above)
Comment (devYaoYH) [reply tools/create_component.py:73]: abstracted into function

File: tools/create_component.py (hunk 6 above)
Comment (devYaoYH) [reply tools/create_component.py:76]: template for index added (originally my though was this file is likely to be static and quite short, doesn't need a mutable template)

**Comment by BhavyaLight** [comment]
why did the files changed rise to 102 ?

**Comment by devYaoYH** [comment]
@redhoteggplant Probably due to redacting keys from our git history?

File: client/template.test.tsx [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
```
Comment (cszqwe) [inline client/template.test.tsx:1]: Consider moving the template to some other folder under client or even in tools as it is actually not part of client code.

File: client/template.tsx [hunk 8]
```diff
@@ -0,0 +1,7 @@
+import React, { useEffect, useState } from "react";
```
Comment (cszqwe) [inline client/template.tsx:1]: Same as above.

File: client/template.test.tsx (hunk 7 above)
Comment (devYaoYH) [reply client/template.test.tsx:1]: I was considering to take some time and explore making this tool a node module of sorts, so then these templates will live on within the 'CreateReactComponent' module files and allow projects to specify non-default templates with a `crc.config.json` file. Will adjust things around in another PR with issue #153

File: client/template.tsx (hunk 8 above)
Comment (devYaoYH) [reply client/template.tsx:1]: will resolve in #153

**Pull request merged**